src/test/java/api/loans/CheckInByBarcodeTests.java

```diff
@@ -149,6 +153,8 @@ public void canCloseAnOpenLoanByCheckingInTheItem() {
 
     assertThat("Checkin Service Point Id should be stored.",
       storedLoan.getString("checkinServicePointId"), is(checkInServicePointId));
+
+    verifyCheckInOperationRecorded(nod.getId(), checkInServicePointId);
```

Comment:
Given that there are unit like tests for this, I think the API tests could demonstrate this behaviour in the simplest possible case e.g. a check in when the item is not borrowed. Rather than attempting to count the occurrences and compare when they occurred.

What do you think?